hudi-utilities/src/main/java/org/apache/hudi/utilities/deltastreamer/HoodieDeltaStreamer.java

```diff
@@ -653,13 +659,14 @@ public static class DeltaSyncService extends HoodieIngestionService {
 
     private final Option<ConfigurationHotUpdateStrategy> configurationHotUpdateStrategyOpt;
 
-    public DeltaSyncService(Config cfg, JavaSparkContext jssc, FileSystem fs, Configuration conf,
+    public DeltaSyncService(Config cfg, HoodieSparkEngineContext sparkEngineContext, FileSystem fs, Configuration conf,
                             Option<TypedProperties> properties) throws IOException {
       super(HoodieIngestionConfig.newBuilder()
           .isContinuous(cfg.continuousMode)
           .withMinSyncInternalSeconds(cfg.minSyncIntervalSeconds).build());
       this.cfg = cfg;
-      this.jssc = jssc;
+      this.sparkEngineContext = sparkEngineContext;
+      this.jssc = sparkEngineContext.getJavaSparkContext();
       this.fs = fs;
```

Comment:
Do we need to keep 2 engine contexts as members ?

Reply:
good call, fixed